### PR TITLE
System Check Script Final PR

### PR DESCRIPTION
**Pull request by yashika-batra**
Note on the structure of the tests: the base level check of “is it there” is run when the satellite object is initialized within pycubed. The results of this check populate the “hardware_dict” that’s referenced in main.py. Every other file / test first checks the hardware_dict to ensure that the object being tested is in fact there, and then runs the higher level “is it functional” check.
- Every “is it functional” test, apart from radio and burnwire, is fairly fast (5 minutes total), so I took your suggestion on only explicitly asking the user if they want to run the radio and burnwire test
- Let me know if there’s any other user input changes y’all want to add (given this setup) / if there’s any structural changes you want me to make to change this

### CODE REVIEW

**Comment by yashika-batra** [comment]
Order to merge system check script in:
First, merge System Check PRs #1-7 with this branch
Then, merge PR #8 (this) with the main branch

File: state_machine/applications/system-check/main.py
Comment (thetazero) [inline state_machine/applications/system-check/main.py:19]: What is the purpose of this?

File: state_machine/applications/system-check/main.py
Comment (thetazero) [inline state_machine/applications/system-check/main.py:93]: Could we make it easier to find which tests have failed and which tests have succeeded?

**Review by thetazero — CHANGES_REQUESTED** [review]
Looks solid, just a little confused about a few things.

Also I'd like to be able to test out the UX of this, could you make it so that if I add the updated driver to this branch I could test out this check script?

File: state_machine/applications/system-check/main.py
Comment (yashika-batra) [reply state_machine/applications/system-check/main.py:19]: The hardware dictionary tells us what hardware has been initialized successfully and what hasn't. It's necessary so we can choose which tests to run and which note, but we don't have to print it (error messages for failed initializations are already printed)

The result dictionary just contains all the different tests as keys, as well as a tuple of a string and boolean as the entry (the string is the log result of the test, and the boolean is True/False for pass/fail)

File: state_machine/applications/system-check/main.py
Comment (yashika-batra) [reply state_machine/applications/system-check/main.py:93]: Yep, will do!

File: state_machine/applications/system-check/main.py
Comment (jbwillis) [inline state_machine/applications/system-check/main.py:69]: Comment says "ask to test" but code doesn't prompt for input.

File: state_machine/applications/system-check/main.py
Comment (jbwillis) [inline state_machine/applications/system-check/main.py:56]: device addresses are not returned. Perhaps change the comment to `print all devices...`

File: state_machine/applications/system-check/main.py
Comment (jbwillis) [inline state_machine/applications/system-check/main.py:49]: When I read this line I expect something to be run on the next line of code. Let's delete `Running` from this line. Also, to make it easier to find in the terminal, let's have this line be

`\n##########    S Y S T E M     C H E C K    ##########\n`

File: state_machine/applications/system-check/main.py
Comment (jbwillis) [inline state_machine/applications/system-check/main.py:50]: I don't love that this line prints "Successful" every time even if something wasn't successful. Delete `Successfully`.

Let's make this:
```
print("Hardware initialization results:")
```

File: state_machine/applications/system-check/main.py
Comment (jbwillis) [inline state_machine/applications/system-check/main.py:51]: nitpick, but you don't need the empty string.

**Review by jbwillis — CHANGES_REQUESTED** [review]

File: state_machine/applications/system-check/main.py
Comment (yashika-batra) [reply state_machine/applications/system-check/main.py:56]: fixed, good catch!

File: state_machine/applications/system-check/main.py
Comment (yashika-batra) [reply state_machine/applications/system-check/main.py:69]: That's probably left over from earlier -- we decided not to ask to test the coil drivers, since it was a straightforward and fast test

**Comment by yashika-batra** [comment]
> Looks solid, just a little confused about a few things.
> 
> Also I'd like to be able to test out the UX of this, could you make it so that if I add the updated driver to this branch I could test out this check script?

Hey @thetazero, sorry I missed this comment. The check script already works with the updated driver. I'm holding off on adding all the files to this branch because I want to make sure we can look at each of them separately.

If you'd like, I can either create a new branch for the current version of the full check script or send a folder to you on Slack for you to test out.

**Review by jbwillis — APPROVED** [review]

**Review by jbwillis — CHANGES_REQUESTED** [review]
Gonna keep this on request changes until all the other test scripts and `main` are merged into this. Then we should test it out on hardware.

**Comment by yashika-batra** [comment]
> Gonna keep this on request changes until all the other test scripts and `main` are merged into this. Then we should test it out on hardware.

Agreed. Regarding testing, I've tested this and sent Losha a copy of the full script to test out. He suggested a few style changes apart from the ones in the PRs, so I'll implement all of that and then re-request review. I can also send you a copy of the folder if you'd like to test it out before we merge everything!

**Comment by yashika-batra** [comment]
Major changes:
- Print the tests that weren't run separately

**Comment by yashika-batra** [comment]
New changes: edited main.py to support the logging infrastructure

**Review by jbwillis — APPROVED** [review]

**Review by jbwillis — CHANGES_REQUESTED** [review]
We need to wait to merge this until all other test things are merged into this.

**Comment by yashika-batra** [comment]
I've tested the IMU and the entire script; it should be ready for UX testing now.

**Review by thetazero — CHANGES_REQUESTED** [review]

**Review by thetazero — APPROVED** [review]

**Review by jbwillis — APPROVED** [review]

**Pull request merged**
